tests/test_oauth2_authorization_code.py

```diff
@@ -145,6 +145,137 @@ def test_refresh_token(token_cache, responses: RequestsMock, browser_mock: Brows
+    )
+
+
+def test_refresh_token_access_token_not_expired(token_cache, responses: RequestsMock, browser_mock: BrowserMock):
+    auth = requests_auth.OAuth2AuthorizationCode(
+        "http://provide_code", "http://provide_access_token"
+    )
+    tab = browser_mock.add_response(
+        opened_url="http://provide_code?response_type=code&state=163f0455b3e9cad3ca04254e5a0169553100d3aa0756c7964d897da316a695ffed5b4f46ef305094fd0a88cfe4b55ff257652015e4aa8f87b97513dba440f8de&redirect_uri=http%3A%2F%2Flocalhost%3A5000%2F",
+        reply_url="http://localhost:5000#code=SplxlOBeZQQYbYS6WxSbIA&state=163f0455b3e9cad3ca04254e5a0169553100d3aa0756c7964d897da316a695ffed5b4f46ef305094fd0a88cfe4b55ff257652015e4aa8f87b97513dba440f8de",
+    )
+    responses.add(
+        responses.POST,
+        "http://provide_access_token",
+        json={
+            "access_token": "2YotnFZFEjr1zCsicMWpAA",
+            "token_type": "example",
+            "expires_in": 3600,
+            "refresh_token": "tGzv3JOkF0XG5Qx2TlKWIA",
+            "example_parameter": "example_value",
+        },
+        match=[
+            urlencoded_params_matcher({
+                "grant_type": "authorization_code",
+                "redirect_uri": "http://localhost:5000/",
+                "response_type": "code",
+                "code": "SplxlOBeZQQYbYS6WxSbIA"
+            })
+        ]
+    )
+    assert (
+            get_header(responses, auth).get("Authorization")
+            == "Bearer 2YotnFZFEjr1zCsicMWpAA"
+    )
+    assert (
+            get_request(responses, "http://provide_access_token/").body
+            == "grant_type=authorization_code&redirect_uri=http%3A%2F%2Flocalhost%3A5000%2F&response_type=code&code=SplxlOBeZQQYbYS6WxSbIA"
+    )
+    tab.assert_success(
+        "You are now authenticated on 163f0455b3e9cad3ca04254e5a0169553100d3aa0756c7964d897da316a695ffed5b4f46ef305094fd0a88cfe4b55ff257652015e4aa8f87b97513dba440f8de. You may close this tab."
+    )
+    # response for refresh token grant
+    responses.add(
+        responses.POST,
+        "http://provide_access_token",
+        json={
+            "access_token": "rVR7Syg5bjZtZYjbZIW",
+            "token_type": "example",
+            "expires_in": 3600,
+            "refresh_token": "tGzv3JOkF0XG5Qx2TlKWIA",
+            "example_parameter": "example_value",
+        },
+        match=[
+            urlencoded_params_matcher(
+                {"grant_type": "refresh_token", "refresh_token": "tGzv3JOkF0XG5Qx2TlKWIA", "response_type": "code"})
+        ]
+    )
+
+    # expect Bearer token to remain the same
+    assert (
+            get_header(responses, auth).get("Authorization")
+            == "Bearer 2YotnFZFEjr1zCsicMWpAA"
+    )
+    # refresh request is not used
+    responses.assert_all_requests_are_fired = False
```

Comment:
Do not add the mock and remove this "hack" instead then ;)

Reply:
Good point, removed the unused response mocks.